addons/event_booth/models/event_booth.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class EventBooth(models.Model):
+    _name = 'event.booth'
+    _description = 'Event Booth'
+    _inherit = [
+        'event.type.booth',
+        'mail.thread',
+        'mail.activity.mixin'
+    ]
+
+    # owner
+    event_type_id = fields.Many2one(ondelete='set null', required=False)
+    event_id = fields.Many2one('event.event', string='Event', ondelete='cascade', required=True)
+    # customer
+    partner_id = fields.Many2one('res.partner', string='Renter', tracking=True, copy=False)
+    contact_name = fields.Char('Renter Name', compute='_compute_contact_name', readonly=False, store=True, copy=False)
+    contact_email = fields.Char('Renter Email', compute='_compute_contact_email', readonly=False, store=True, copy=False)
+    contact_mobile = fields.Char('Renter Mobile', compute='_compute_contact_mobile', readonly=False, store=True, copy=False)
+    contact_phone = fields.Char('Renter Phone', compute='_compute_contact_phone', readonly=False, store=True, copy=False)
+    # state
+    state = fields.Selection(
+        [('available', 'Available'), ('unavailable', 'Unavailable')],
+        string='Status', group_expand='_group_expand_states',
+        default='available', required=True, tracking=True,
+        help='Shows the availability of a Booth')
+    is_available = fields.Boolean(compute='_compute_is_available', search='_search_is_available')
+
+    @api.depends('partner_id')
+    def _compute_contact_name(self):
+        for booth in self:
+            if not booth.contact_name:
+                booth.contact_name = booth.partner_id.name or False
+
+    @api.depends('partner_id')
+    def _compute_contact_email(self):
+        for booth in self:
+            if not booth.contact_email:
+                booth.contact_email = booth.partner_id.email or False
+
+    @api.depends('partner_id')
+    def _compute_contact_mobile(self):
+        for booth in self:
+            if not booth.contact_mobile:
+                booth.contact_mobile = booth.partner_id.mobile or False
+
+    @api.depends('partner_id')
+    def _compute_contact_phone(self):
+        for booth in self:
+            if not booth.contact_phone:
+                booth.contact_phone = booth.partner_id.phone or False
+
+    @api.depends('state')
+    def _compute_is_available(self):
+        for booth in self:
+            booth.is_available = booth.state == 'available'
+
+    def _search_is_available(self, operator, operand):
+        negative = operator in expression.NEGATIVE_TERM_OPERATORS
+        if (negative and operand) or not operand:
+            return [('state', '=', 'unavailable')]
+        return [('state', '=', 'available')]
```

Comment:
is this field really necessary ? as basically = state == available. i guess we could replace is_available by state == available everywhere, add filter "is available" if needed. That would save a field + a search method. Or do we have a more complex usage for it ?